File: Data Structures and Algorithms/4. Recursion/Kunal/Level 1/_4_sum_of_digits.java
import java.util.*;

public class _4_sum_of_digits{
    public static int sum(int n){
        int s = 0;
        while(n != 0) {
            int last_digit = n % 10;
            s += last_digit;
            n /= 10;
        }
        return s;
    }

    public static int recursive(int n, int s) {
        if(n == 0) return s;
        int last_digit = n % 10;
        return recursive(n /= 10, s += last_digit);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        System.out.println("Sum of digits of number " + n + " is " + recursive(n, 0));
    }
}

File: Data Structures and Algorithms/4. Recursion/Kunal/Level 1/_4_sum_of_digits.cpp
#include <iostream>
using namespace std;

int sum(int n, int s) {
    if(n == 0)
        return s;

    int last_digit = n % 10;
    return sum(n /= 10, s += last_digit);
}

int recursive(int n) {
    if(n == 0)
        return 0;
    return (n % 10 + recursive(n / 10));
}

int main() {
    int n;
    cin >> n;

    cout << "Sum of digits " << n << " is " << recursive(n) << '\n';
    return 0;
}
